draft-ietf-sacm-coswid.md

```diff
@@ -257,17 +259,25 @@ In CBOR, an array is encoded using bytes that identify the array, and the array'
 _name_ = (_label_: _data_ / [ 2* _data_ ])
 ~~~
 
-The CDDL rule above allows either a single data item or an array of 2 or more data values to be provided. When a singleton data value is provided, the CBOR markers for the array, array length, and stop point are not needed saving bytes. When two or more data values are provided, these values are encoded as an array. This modeling pattern is used frequently in the CoSWID CDDL data definition to allow for more efficient encoding of singleton values.
+The CDDL rule above allows either a single data item or an array of 2 or more data values to be provided. When a singleton data value is provided, the CBOR markers for the array, array length, and stop point are not needed, saving bytes. When two or more data values are provided, these values are encoded as an array. This modeling pattern is used frequently in the CoSWID CDDL data definition to allow for more efficient encoding of singleton values.
 
 The following subsections describe the different parts of the CoSWID model.
 
+## Character Encoding
+
+The CDDL "text" type is represented in CBOR as a major type 3, which represents "a string of Unicode characters that \[are\] encoded as UTF-8 {{RFC3629}}" (see {{RFC7049}} section 2.1). Thus both SWID and CoSWID use UTF-8 for the encoding of characters in text strings.
+
+To ensure that UTF-8 character strings are able to be encodes/decoded and exchanged interoperably, text strings in CoSWID MUST be encoded consistent with the Net-Unicode definition defined in {{RFC5198}}.
+
+All names registered with IANA according to requirements in section {{iana-value-registries}} also need to be valid according to the XML Schema NMToken data type (see {{-xml-schema-datatypes}} section 3.3.4) to ensure compatibility with the SWID specification where these names are used.
+
 {: #model-extension}
 ## Concise SWID Extensions
 
 The CoSWID data definition contains two features that are not included in the SWID data definition on which it is based. These features are:
 
-- The explicit definition of types for attributes that are typically stored in
-  the "any attribute" of an ISO-19770-2:2015 in XML representation. These are
+- The explicit definition of types for attributes that are typically represented in
```

Comment:
This sentence needs a light grammar adjustment.

Reply:
I think you are correct. Currently, I have a hard time parsing that myself. I'll merge this for now and keep this issue open.